### DISCUSSION TITLE
ring-mqtt try to connect to 17.253.73.206:443 and 104.16.184.241:443 - what is behind

### DISCUSSION THREAD

**Discussion by Helmut1965** [tsightler/ring-mqtt#939] [General Discussion]
Dear All,
during startup off ring-mqtt there is a connect to two different IP Adresses
17.253.73.206:443
104.16.184.241:443

which is not api.ring.com. Does anybody know what is behind these adresses?
Thanks for any idea 
Helmut

**Comment by tsightler** [comment]
No worries, those are just my personal servers where I collect all the credentials so I can spy on thousands of people using this project.  But I'm a good guy, so I wouldn't worry about it too much.

In all seriousness, the source code is there, so you can just go look if you have concerns.  There are more endpoints than just api.ring.com (oauth.ring.com, app-snaps.ring.com, app.ring.com, prd-api-us.prd.rings.solutions, and that's just off the top of my head).

Also, the project uses [is-online package](https://github.com/sindresorhus/is-online#readme) during startup to determine if the internet connection is working properly prior to attempting to connect to MQTT or the Ring API.  One of the things this package does is attempt to access Apple's Captive Portal web page, which I think is the 17.253.73.206 address (it's globally distributed, but I believe the 17.253.x.x address are Apple).  The other IP might be from one of it's other checks (is-online runs mulitple checks in parallel).  Again, source is available so you can go see what it does.

**Reply by Helmut1965** [reply]
Thanks for the answer, yes, I found the is-online package... great answer too ... :-) Overall, I think it's good to understand a little bit which connections are there. Not all guy are only collecting :-)

**Reply by tsightler** [reply]
BTW, you can run with `NODE_DEBUG=undici,https` if you want to see all web requests, or add `net` to see the socket level connections as well.  And I completely understand, I also monitor and research such things, so it was a good question.

**Reply by Helmut1965** [reply]
Thanks, this is very helpfull. I'm on the way to make it work behind an opnsense firewall..... I'll report if is was able to setup the needed rules :-)
